CMakeLists.txt

```diff
@@ -238,6 +238,8 @@ set_target_properties(openPMD PROPERTIES
     CXX_EXTENSIONS OFF
     CXX_STANDARD_REQUIRED ON
     POSITION_INDEPENDENT_CODE ON
+    CXX_VISIBILITY_PRESET hidden
```

Comment:
for consistency with the `Visibility.hpp`  header we should check here:
- is this compiler MSVC or
- does this compiler have support for `__attribute__((visibility("default")))`

then hide, otherwise don't.

Reply:
Maybe we can check via `$<COMPILE_FEATURES:cxx_attributes>`

https://cmake.org/cmake/help/v3.1/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html#prop_gbl:CMAKE_CXX_KNOWN_FEATURES